audio-worklet/bitcrusher.js

```diff
@@ -0,0 +1,234 @@
+  Run series of tests on Bitcrusher node, 
+*/
+function testBitcrusher() {
+  /* ---------------------------------------------------------------------------- 
+     Test Bitcrusher rounding method. 
+     ----------------------------------------------------------------------------*/
+  let base = 10; // set base to 10 for easier testing
+  let x = Bitcrusher.round(1.7821, 2, base); 
+  let expected = 1.78;
+  console.assert(x == expected, "computed " + x + " but expected " + expected);
+
+  x = Bitcrusher.round(-1.332432, 5, base); 
+  expected = -1.33243;
+  console.assert(x == expected, "computed " + x + " but expected " + expected);
+
+  x = Bitcrusher.round(10.322, 3, base); 
+  expected = 10.322;
+  console.assert(x == expected, "computed " + x + " but expected " + expected);
+
+  x = Bitcrusher.round(23.432123212321232123, 9, base); 
+  expected = 23.432123212;
+  console.assert(x == expected, "computed " + x + " but expected " + expected);
+
+  x = Bitcrusher.round(1, 1, base); 
+  expected = 1;
+  console.assert(x == expected, "computed " + x + " but expected " + expected);
+  /* ---------------------------------------------------------------------------- */
+
+  
+  /* ---------------------------------------------------------------------------- 
+     Test sample reduction component. 
+     ---------------------------------------------------------------------------- */
+  // Hold constant precision and test factor reduction
+  // Populate a Float Array with 0:99 and verify the bitcrusher holds [0, 0, ... 0] [10, 10, ... 10]
+  // where the number of consecutive equal numbers is controlled by factor
+  for (let factor = 1; factor < 10; factor++) {
+    const precision = 1; // leave precision in tact
+    const bufferSize = 100; // arbitrarily sized buffer for testing
+    let referenceBuffer = new Float32Array(bufferSize);
+    let outputBuffer = new Float32Array(bufferSize);
+    
+    for (let i =0; i < bufferSize; i++) {
+      referenceBuffer[i] = i; // populate array 1:100
+    }
+
+    Bitcrusher.bitcrush(factor, precision, referenceBuffer, outputBuffer); 
+    
+    outputBuffer.map(function(d, i) { // d is bitcrushed data and i is index in array
+      let expected = Math.floor(i / factor) * factor;
+      console.assert(d == expected, "computed " + d + " but expected " + expected);
+    });
+  }
+  
+  /*---------------------------------------------------------------------------- 
+    Test Bitcrusher in Offline Web Audio context  
+    ---------------------------------------------------------------------------- */ 
+  // Test that Bitcrusher class works in offline Web Audio context.
+  // Create two oscillators, runing one through a bitcrusher with variables set to avoid any effect.
+  // Conjoin the nodes through a ChannelMergerNode and feed output to an offline destination.
+  // Verify that the samples are nearly identical
+  let offlineContext = new OfflineAudioContext(2, 12800, 12800); // create 2 channels for comparison
```

Comment:
Why 12800 Hz?  (Just add a comment on why you picked this instead of the more typical 44.1 kHz or 48 kHz.)